### PR TITLE
 fix(lint): fix lint errors and add yapf formatter with pre-commit hook

### PR DESCRIPTION
**Pull request by fristonio**
- add pre-commit hook and script to install it.
- format lib with yapf.

### CODE REVIEW

**Comment by uttu357** [comment]
Can you fix this asap?

**Comment by fristonio** [comment]
Yep, in a minute.

**Comment by fristonio** [comment]
@uttu357 Rebased the branch.

File: hooks/Makefile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Install git hooks for development.
+install:
+	@cp ./pre-commit.sh ./../.git/hooks/pre-commit
+	@chmod +x ./../.git/hooks/pre-commit
+
+.PHONY: install
```
Comment (uttu357) [inline hooks/Makefile]: Please check this everytime.. Endlines.. :P

Comment (fristonio) [reply hooks/Makefile]: I have added an editorconfig, so issues like this won't be there from now on.

**Pull request merged**